src/model/SolarCollectorFlatPlatePhotovoltaicThermal.cpp

```diff
@@ -282,16 +300,33 @@ namespace model {
 
     SolarCollectorPerformancePhotovoltaicThermalSimple performance(model);
 
-    bool ok = true;
-    ok = getImpl<detail::SolarCollectorFlatPlatePhotovoltaicThermal_Impl>()->setSolarCollectorPerformanceNoClone(performance);
+    const bool ok = getImpl<detail::SolarCollectorFlatPlatePhotovoltaicThermal_Impl>()->setSolarCollectorPerformanceNoClone(performance);
+    OS_ASSERT(ok);
+  }
+
+  SolarCollectorFlatPlatePhotovoltaicThermal::SolarCollectorFlatPlatePhotovoltaicThermal(
+    const SolarCollectorPerformancePhotovoltaicThermalBIPVT& performance)
+    : StraightComponent(SolarCollectorFlatPlatePhotovoltaicThermal::iddObjectType(), performance.model()) {
+    OS_ASSERT(getImpl<detail::SolarCollectorFlatPlatePhotovoltaicThermal_Impl>());
+
+    const bool ok = getImpl<detail::SolarCollectorFlatPlatePhotovoltaicThermal_Impl>()->setSolarCollectorPerformanceNoClone(performance);
+    OS_ASSERT(ok);
+  }
+
+  SolarCollectorFlatPlatePhotovoltaicThermal::SolarCollectorFlatPlatePhotovoltaicThermal(
```

Comment:
Two new ctors, passing in performance which is either ThermalSimple or ThermalBIPVT.